### PR TITLE
Adds `sample_lognormal()`

### PR DESCRIPTION
**Pull request by kellijohnson-NOAA**
# What is the feature?
* Adds a function to sample a vector and add log normal error to it.

# How have you implemented the solution?
* Using code from ss3sim, a lognormal distribution with a mean of zero and a user-provided standard deviation is simulated the same length as the input vector. These samples are then bias corrected, exponentiated, and multiplied by the input vector, the result is then returned.

# Does the PR impact any other area of the project, maybe another repo?
* No

### CODE REVIEW

File: tests/testthat/test-sample_lognormal.R [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  #' @description Test that sample_lognormal(x) returns two different values
+  #' when the seed is not set.
+  expect_false(
+    sample_lognormal(1) == sample_lognormal(1)
+  )
+})
+
+## Edge handling ----
+test_that("sample_lognormal() returns correct outputs for edge cases", {
+  #' @description Test that sample_lognormal(x) returns a vector when a single
```
Comment (Bai-Li-NOAA) [inline tests/testthat/test-sample_lognormal.R]: Currently, the @description tag only supports a single line. If it spans multiple lines, it can cause misalignment in the report's table cells.

Is a vector being passed to `x` and a single value to `sd` in this case?

File: tests/testthat/test-sample_lognormal.R [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  #' value is passed to `x` but a vector is passed to `sd`.
+  expect_equal(
+    object = length(sample_lognormal(1:5, 1)),
+    expected = 5
+  )
+
+  #' @description Test that sample_lognormal(x) returns a vector when a single
+  #' value is passed to `sd` but a vector is passed to `x`.
```
Comment (Bai-Li-NOAA) [inline tests/testthat/test-sample_lognormal.R]: Currently, the @description tag only supports a single line. If it spans multiple lines, it can cause misalignment in the report's table cells.

Is a single value being passed to `x` and a vector to `sd` in this case?

File: tests/testthat/test-sample_lognormal.R [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  expect_equal(
```
Comment (Bai-Li-NOAA) [inline tests/testthat/test-sample_lognormal.R:59]: Question: If I understand correctly, the user provides a single observation value and a vector of sds. The `stats::rnorm()`  samples one value using only the first `sd`. However, `bias_correction` returns 5 values because the `sd` vector has 5 elements. As a result, the final expected values also have a length of 5. Should we replicate `x` into a vector of length 5 before using `stats::rnorm()`, or should we instead use only the first `sd` when calculating `bias_correction`? Or am I misunderstanding something here?

**Review by Bai-Li-NOAA — CHANGES_REQUESTED** [review]
Thank you, @kellijohnson-NOAA, The function and documentation look really great. I have a question about a scenario involving one observation and a vector of sds, as well as a request to update the documentation.

Is this branch based on `calc_fxns`? Would it be possible to remove the `add calc_truth() `commit and merge the rest into main? Sorry, but `calc_fxns` isn’t ready for a full merge into main yet.

File: tests/testthat/test-sample_lognormal.R (hunk 3 above)
Comment (kellijohnson-NOAA) [reply tests/testthat/test-sample_lognormal.R:59]: Thank you @Bai-Li-NOAA for catching this. I have corrected the function to repeat x to be the same length as sd when length(x) ==1 and length(sd) > 1.

**Comment by kellijohnson-NOAA** [comment]
I can pull out the calc_fxns commit no problem. I am wondering though if it is okay to leave the multi-line description in the tests because we have the fix with the fork of that package to allow for longer descriptions?

File: tests/testthat/test-sample_lognormal.R (hunk 1 above)
Comment (kellijohnson-NOAA) [reply tests/testthat/test-sample_lognormal.R]: Fixed, thank you!

File: tests/testthat/test-sample_lognormal.R (hunk 2 above)
Comment (kellijohnson-NOAA) [reply tests/testthat/test-sample_lognormal.R]: Fixed, thank you!
